memcache/memcache.go

```diff
@@ -89,7 +89,7 @@ func legalKey(key string) bool {
 		return false
 	}
 	for i := 0; i < len(key); i++ {
-		if key[i] <= ' ' || key[i] > 0x7e {
+		if key[i] <= ' ' || key[i] == 0x7f {
```

Comment:
In ASCII they are below 27, which is covered by the first comparison clause. After that, only 127 remains.